### PR TITLE
docs: XML Comments improvements

### PR DESCRIPTION
**Pull request by VaclavElias**
# PR Details

Summaries added based on Bepu source code.

## Related Issue

N/a.

## Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My change requires a change to the documentation.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] **I have built and run the editor to try this change out.**

### CODE REVIEW

**Comment by Eideren** [comment]
Duplicate of https://github.com/stride3d/stride/pull/2567 for all but one case - mine doesn't lift them all sale, I tried to actually explain the utility of those in layman's terms so I would rather keep mine

**Comment by VaclavElias** [comment]
Nice, I should have studied more existing PRs 🤦‍♂️. Never mind. We can close this and you can take anything you need from `SpeculativeMargins`. I think, it's definitelly better as you do it, reworded and in layman's terms. Thanks. Didn't want to step on your toes 🤣.

I might ping Norbo, if he would like a help generating Bepu APIs to easier access all the features Bepu can offer but mainly we can direct readers also to the original docs if someone would like to dig deeper. Thanks.

**Comment by Eideren** [comment]
Thanks !
